### PR TITLE
Exists in nested collections

### PR DESCRIPTION
**Pull request by paf31**
[Rendered RFC](https://github.com/hasura/ndc-spec/blob/feature/exists-in-nested-collection/rfcs/0020-exists-in-nested-collection.md)

---

- [x] RFC
- [x] Specification updates
  - [ ] Changelog
  - [x] Specification pages
  - [ ] Tutorial pages
  - [x] `reference/types.md`
- [x] Implement your feature in the reference connector
- [ ] Generate test cases in `ndc-test` if appropriate
- [x] Does your feature add a new capability?
  - [x] Update `specification/capabilities.md` in the specification
  - [ ] Check the capability in `ndc-test`

### CODE REVIEW

File: ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                "type": "nested_collection",
+                "arguments": {
+                    "limit": {
+                        "type": "literal",
+                        "value": null
+                    }
+                },
+                "column_name": "staff"
+            },
+            "predicate": {
```
Comment (danieljharvey) [inline ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json:38]: Just checking my understanding - right now the example filters `last_name` containing "s", if we wanted to check for `specialities` that include an "s" instead, we'd nest another `exists` with a `nested_collection` here and put the `binary_comparison_operator` inside that?

File: ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                "operator": "like",
+                "value": {
+                    "type": "scalar",
+                    "value": "s"
+                }
+            }
+        }
+    },
+    "collection_relationships": {
+        "author_articles": {
```
Comment (danieljharvey) [inline ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json:53]: Am I missing something, or is this relationship definition not required for this request?

**Review by danieljharvey — APPROVED** [review]
A couple of clarifying questions for my understanding, this looks like it will do what we need though, great!

File: ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json (hunk 1 above)
Comment (paf31) [reply ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json:38]: Correct, but we can only use `exists` on nested _collections_, not nested arrays of strings. You would need to use something like a `contains` operator for your other example.

File: ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json (hunk 2 above)
Comment (paf31) [reply ndc-reference/tests/query/predicate_with_exists_in_nested_collection/request.json]: Yes, thanks.

**Pull request merged**